pkg/satellite/validation_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellite
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidationUserInfo(t *testing.T) {
+	cases := []struct {
+		testName string
+		in       UserInfo
+		out      error
+	}{
+		{
+			testName: "Valid test 1",
+			in: UserInfo{
```

Comment:
Use a one liner for declaring user info. You can use a local func for it.

```
    type Case struct { User UserInfo; Errors []error }
    testcase := func(first, last, pass, email string, errs ...error)  Case {
        return Case {
            ...
        }
    }

    var cases []Case{
        testcase("Bill", "", "123456789", "a@b.com"),
        testcase("Bill", "", "123456789", "a@b.com", ErrMissingField("something")),
        ...
    }

    for _, test := range cases {
        t.Run(fmt.Sprintf("%+v", test.User), func(t *testing.T) {
            // ...
        })
    }
```

Although here is better not to use subtests as it would clutter the test output a lot. It's better to get it showing up as a single test.

```
    for _, test := range cases {
        testname := fmt.Sprintf("%+v", test.User)
        ...
        if !ok {
            t.Errorf("%v: %v", testname, errortext)
        }
    }
```